### PR TITLE
support logprobs for spec v2

### PR DESCRIPTION
**Pull request by Ratish1**
## Motivation

   This PR implements overlap-safe logprob support for spec v2 (EAGLE) by following the forward/scheduler split.

 Spec v2 did not compute output logprobs in eagle_worker_v2.verify() and the scheduler decode output path did not collect spec-v2 flattened logprob payload per request. This caused logprob failures for spec v2 requests. #19761 #17349

## Modifications

  1. In eagle_worker_v2.verify(), after sample() returns predict/accept_length/accept_index, compute accepted-token logprobs on GPU using log_softmax and accepted-index gather, then populate:
 
  - logits_output.next_token_logprobs
  - logits_output.next_token_top_logprobs_val
  - logits_output.next_token_top_logprobs_idx
 
  2. In scheduler decode output processing (process_batch_result_decode), add spec-v2 logprob collection from CPU-copied logits output using a per-request pointer over flattened accepted-token outputs.
  3. Ensure draft prefill and draft-extend paths do not request logits-processor logprob computation; spec-v2 returned logprobs are produced in verify as designed.
  4. Add test coverage for OpenAI completions logprobs in spec-v2 EAGLE test suite.

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

File: test/registered/spec/eagle/test_eagle_infer_beta.py [hunk 1]
```diff
@@ -92,6 +94,35 @@ def test_gsm8k(self):
...
+                "logprobs": 1,
+            },
+        )
+        self.assertEqual(response.status_code, 200, msg=response.text)
+
+        response_json = response.json()
+        self.assertIn("choices", response_json)
+        self.assertGreater(len(response_json["choices"]), 0)
+
+        logprobs = response_json["choices"][0]["logprobs"]
```
Comment (hnyls2002) [inline test/registered/spec/eagle/test_eagle_infer_beta.py:115]: You only check the length, not the accuracy? Please compare this to spec v1 / normal cases

**Review by hnyls2002 — CHANGES_REQUESTED** [review]

File: python/sglang/srt/speculative/eagle_worker_v2.py [hunk 2]
```diff
@@ -784,6 +790,44 @@ def verify(self, batch: ModelWorkerBatch):
...
+                        batch.top_logprobs_nums,
+                        accept_length.tolist(),
+                        strict=True,
+                    )
+                    for _ in range(num_tokens)
+                ]
+                (
+                    logits_output.next_token_top_logprobs_val,
+                    logits_output.next_token_top_logprobs_idx,
+                ) = get_top_logprobs(logprobs, top_logprobs_nums_repeat_interleaved)
```
Comment (Qiaolin-Yu) [inline python/sglang/srt/speculative/eagle_worker_v2.py:827]: I think here it will incur extra gpu->cpu sync?

**Review by Qiaolin-Yu — COMMENTED** [review]
Thanks for your contribution. This pr may have small issues. I opened another pr too #19801. I will combine these two PRs into one and add you as a co-author in that pr.

**Comment by Qiaolin-Yu** [comment]
btw, what's your github email?
